app/src/main/admin/components/AdminApp.tsx

```diff
@@ -33,4 +33,12 @@ export class AdminAppComponent extends React.Component<AdminAppProps, undefined>
     }
 }
 
-export const AdminApp = connectToStores(AdminAppComponent);
\ No newline at end of file
+export const AdminAppAltWrapped = connectToStores(AdminAppComponent);
+
+const mapStateToProps = (state: any) => {
+    return {
+        loggedIn: state.auth.loggedIn,
```

Comment:
I think ideally we'd have some type checking here. The ideas I have are
1. first set `const auth: AuthState = state.auth`
2. have a `GlobalState` interface with all possible state properties across the 3 apps, then this method signature can be
```
const mapStateToProps = (state: Partial<GlobalState>) => {
}
```

I prefer option 2. What does everyone else think?

Reply:
TypeScript allows type compositions, right? So we could have something even better than option 2, along the lines of:

```
type GlobalState = ContribState | AdminState | ReportState
```

Those app-specific types could, of course, have a shared based interface, for things that are useful across all apps, if any.

Reply:
Ah, nice idea.